gnomad_qc/v5/sample_qc/create_sample_qc_metadata_ht.py

```diff
@@ -0,0 +1,571 @@
+
+    :param meta_ht: Table with metadata.
+    :param samples_to_exclude: Expression with samples to exclude.
+    :return: Annotated meta Table with hard-filter fields.
+    """
+    # Add AoU hard filters hard filters and samples to exclude and build hard-filters annotation by project.
+    # Note: Verified that none of these excluded sample IDs overlap with
+    # gnomAD sample IDs, so no need to call 'add_project_prefix_to_sample_collisions' function.
+    is_excluded = samples_to_exclude.contains(meta_ht.s)
+    aou_qc_filters = aou_hard_filters_ht[meta_ht.s].sample_qc_metric_hard_filters
+    has_qc_filters = hl.is_defined(aou_qc_filters) & (hl.len(aou_qc_filters) > 0)
+    meta_ht = meta_ht.annotate(
+        hard_filters=(
+            hl.case()
+            .when(
+                meta_ht.project_meta.project == "gnomad",
+                hl.or_else(meta_ht.hard_filters, hl.empty_set(hl.tstr)),
+            )
+            .when(
+                meta_ht.project_meta.project == "aou",
+                hl.empty_set(hl.tstr)
+                .union(
+                    hl.if_else(has_qc_filters, aou_qc_filters, hl.empty_set(hl.tstr))
+                )
+                .union(
+                    hl.if_else(
+                        is_excluded, hl.set(["sample_exclusion"]), hl.empty_set(hl.tstr)
+                    )
+                ),
+            )
+            .default(hl.empty_set(hl.tstr))
+        )
+    )
+
+    return meta_ht
+
+
+def annotate_genetic_ancestry(
+    meta_ht: hl.Table, v4_meta_ht: hl.Table, aou_gen_anc_ht: hl.Table
+) -> hl.Table:
+    """
+    Annotate metadata Table with genetic ancestry information.
+
+    .. note::
+
+        - For 'gnomad' project, uses v4 genetic ancestry inference.
+        - For 'aou' project, uses AoU projected genetic ancestry.
+
+    :param meta_ht: Table with metadata.
+    :param v4_meta_ht: gnomAD v4 metadata Table containing inferred genetic ancestry.
+    :param aou_gen_anc_ht: Table with AoU projected genetic ancestry.
+    :return: Table annotated with genetic ancestry inference outputs.
+    """
+    # Add "gnomad" prefix to gnomAD v4 meta Table.
+    v4_meta_ht = add_project_prefix_to_sample_collisions(
+        t=v4_meta_ht,
+        sample_collisions=sample_id_collisions.ht(),
+        project="gnomad",
+    )
+
+    # Nest genetic ancestry inference fields under
+    # 'genetic_ancestry_inference' and drop unnecessary fields.
+    v4_meta_ht = v4_meta_ht.annotate(
+        genetic_ancestry_inference=hl.struct(
+            gen_anc=v4_meta_ht.population_inference.pop,
+            **{
+                k: v
+                for k, v in v4_meta_ht.population_inference.items()
+                if k not in ["pop", "prob_oth", "training_pop_all", "training_pop"]
+            },
+        )
+    )
+
+    # Align AoU fields with v4 fields.
+    v4_fields = [f for f in v4_meta_ht.genetic_ancestry_inference.dtype.fields]
+    fields_exprs = {f: aou_gen_anc_ht[f] for f in v4_fields}
```

Comment:
is this just to ensure the fields are in the same order? it seems like they would be already based on the table schema
```
----------------------------------------
Global fields:
    'assign_gen_ancs_from_pc_params': struct {
        min_assignment_prob: float64
    } 
    'min_prob_cutoffs': dict<str, float64> 
----------------------------------------
Row fields:
    's': str 
    'pca_scores': array<float64> 
    'gen_anc': str 
    'prob_afr': float64 
    'prob_ami': float64 
    'prob_amr': float64 
    'prob_asj': float64 
    'prob_eas': float64 
    'prob_fin': float64 
    'prob_mid': float64 
    'prob_nfe': float64 
    'prob_sas': float64 
----------------------------------------
Key: ['s']
----------------------------------------
```